queries/gchain_citizens.sql

```diff
@@ -0,0 +1,54 @@
+with
+-- For a permanent version of this query please visit: https://dune.xyz/queries/453088
+investment_claims as (
+    select concat('0x', encode(claimant, 'hex'))   as claimant,
+           sum(case when "claimType" = 1 then "claimedAmount" else 0 end)                                  as gno_option_claimed,
+           sum(case when "claimType" = 2 then "claimedAmount" else 0 end)                                  as user_option_claimed,
+           sum(case when "claimType" = 1 then "claimableAmount" end)                                       as claimable_gno_option,
+           sum(case when "claimType" = 2 then "claimableAmount" end)                                       as claimable_user_option,
+           -- claim indices are constructed of block_time * 10^5 + evt_index (giving a unique ordering on the time of investment)
+           -- Note that event indices for this table are all strictly less than 10^5 (cf https://dune.xyz/queries/453401)
+           sum(case when "claimType" = 1 then EXTRACT(EPOCH FROM evt_block_time) * 10 ^ 5 + evt_index end) as gno_claim_index,
+           sum(case when "claimType" = 2 then EXTRACT(EPOCH FROM evt_block_time) * 10 ^ 5 + evt_index end) as user_claim_index
+    from cow_protocol."CowProtocolVirtualToken_evt_Claimed"
+    group by claimant
+),
+
+independent_investments_exercised as (
+    select claimant,
+           gno_claim_index,
+           user_claim_index,
+           case
+               when claimable_gno_option is not null then gno_option_claimed / claimable_gno_option
+           end                                                                                           as gno_option_exercised,
+           case
+               when claimable_user_option is not null then user_option_claimed / claimable_user_option
+           end                                                                                           as user_option_exercised,
+           case
+               when greatest(gno_option_claimed, user_option_claimed) = gno_option_claimed then 'GNO'
+               when greatest(gno_option_claimed, user_option_claimed) = user_option_claimed then 'xDAI'
+           end                                                                                           as dominant_claim
+    from investment_claims
+    where gno_option_claimed + user_option_claimed > 0 --! Exclude all candidates who did not invest at all or were not eligible
+),
```

Comment:
It's not really clear to me how you decide which investment belongs to `GC` and which to `Ethereum` since there isn't any obvious clause in the query like `where chain = GC` and the `claimType`s are also not unique to the chain.
I'm guessing the highlighted line and the equivalent line on the other query would have the effect that every investment can only be associated to one chain. 
If that is the case, could you elaborate a little bit how that works or point me to the right piece of code if that is unrelated?

Reply:
There are two separate queries on each network (dune does not allow cross chain queries). In the python code, the distinction is specified in `dune.fetch` method.

Reply:
I saw that but when I manually tried to query some data on the table`cow_protocol."CowProtocolVirtualToken_evt_Claimed"` it didn't seem to make a difference when I switched between `GC` and `Ethereum` so I thought there must be something implicit about it. Maybe I didn't reload the page properly. 🤷‍♂️ 
In any case that was the reason for my confusion but it's now resolved. 👍